pyquil/paulis.py

```diff
@@ -133,7 +134,7 @@ def __init__(
         if op not in PAULI_OPS:
             raise ValueError(f"{op} is not a valid Pauli operator")
 
-        self._ops: Dict[int, str] = OrderedDict()
+        self._ops: Dict[PauliTargetDesignator, str] = OrderedDict()
```

Comment:
I think the return type of `operations_as_set()` should probably be updated to reflect this?

I also don't think this type (and by extension the type for the `index` argument above) is quite right. I think it should include `QubitPlaceholder`? See e.g. `pyquil/tests/test_paulis_with_placeholders.py`

Reply:
> I think the return type of operations_as_set() should probably be updated to reflect this?

`operations_as_set()` return type updated in the 2nd commit.

> I think it should include QubitPlaceholder?

Do I update the `index` to be `Union[PauliTargetDesignator, QubitPlaceholder]` or do I change the definition of `PauliTargetDesignator` to include `QubitPlaceholder`?

Reply:
I think adding `QubitPlaceholder` to `PauliTargetDesignator` makes the most sense, assuming it doesn't break other type checking assumptions.

Reply:
> I think adding QubitPlaceholder to PauliTargetDesignator makes the most sense, assuming it doesn't break other type checking assumptions.

Added in 3rd commit.